bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,285 @@
+            fetchNewPrices();
+            return;
+        }
+
+        boolean needsInitialUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, cetZoneId))
+                || needToFetchHistoricDays(true);
+
+        boolean hasNextDayValue = responseMap.entrySet().stream()
+                .anyMatch(x -> x.getKey().getDayOfYear() == currentUtcTime().plusDays(1).getDayOfYear());
+
+        boolean readyForNextDayValue = currentCetTime()
+                .isAfter(currentCetTime().withHour(config.spotPricesAvailableCetHour));
+
+        if (needsInitialUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            fetchNewPrices();
+        } else {
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            schedule(true);
+        }
+    }
+
+    private void fetchNewPrices() {
+        logger.trace("Fetching new prices");
+        ZonedDateTime startUtc = currentUtcTimeWholeHours()
+                .minusDays(needToFetchHistoricDays() ? config.historicDays : 1).withHour(22);
+        ZonedDateTime endUtc = currentUtcTimeWholeHours().plusDays(1).withHour(22);
+
+        Request request = new Request(config.securityToken, config.area, startUtc, endUtc);
+        Client client = new Client();
+        boolean success = false;
+
+        try {
+            responseMap = client.doGetRequest(request, 30000);
+            TimeSeries baseTimeSeries = new TimeSeries(EntsoeBindingConstants.TIMESERIES_POLICY);
+            for (Map.Entry<ZonedDateTime, Double> entry : responseMap.entrySet()) {
+                BigDecimal kwhPrice = BigDecimal.valueOf(entry.getValue()).divide(new BigDecimal(1000), 10,
+                        RoundingMode.HALF_UP);
+                baseTimeSeries.add(entry.getKey().toInstant(), getPriceState(kwhPrice));
+            }
+            updateStatus(ThingStatus.ONLINE);
+            lastDayAheadReceived = currentCetTime();
+            sendTimeSeries(EntsoeBindingConstants.CHANNEL_SPOT_PRICE, baseTimeSeries);
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            scheduler.schedule(this::triggerChannelSpotPricesReceived, 0, TimeUnit.SECONDS);
+            success = true;
+        } catch (EntsoeResponseException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (EntsoeConfigurationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        } finally {
+            schedule(success);
+        }
+    }
+
+    private State getPriceState(BigDecimal kwhPrice) {
+        if (baseUnit == null || fromUnit == null) {
+            logger.trace("Currency is unknown. Falling back to DecimalType");
+            return new DecimalType(kwhPrice);
+        }
+        try {
+            return new QuantityType<>(kwhPrice + " " + EntsoeBindingConstants.ENTSOE_CURRENCY + "/kWh");
```

Comment:
I had a look at the payload, and it seems we don't need to hardcode the unit:
```xml
<currency_Unit.name>EUR</currency_Unit.name>
```

I don't know if it can be anything other than EUR, but in case they make it user-configurable at some point (i.e. providing currency conversions from the service), this would not be supported when having a hardcoded unit here.

Reply:
agree, implemented